packages/kbn-eslint-plugin-i18n/helpers/get_i18n_identifier_from_file_path.ts

```diff
@@ -14,18 +14,35 @@ export function getI18nIdentifierFromFilePath(fileName: string, cwd: string) {
   const { dir } = parse(fileName);
   const relativePathToFile = dir.replace(cwd, '');
 
-  const relativePathArray = relativePathToFile.split('/');
+  const relativePathArray = relativePathToFile.includes('src')
+    ? relativePathToFile.split('/').slice(1)
+    : relativePathToFile.split('/').slice(2);
 
-  const path = `${relativePathArray[2]}/${relativePathArray[3]}`;
+  const publicFolderIndex = relativePathArray.findIndex((el) => el === 'public' || el === 'server');
+
+  const path = relativePathArray.reduce((acc, curr, index) => {
+    if (index < publicFolderIndex) {
+      return index === 0 ? curr : `${acc}/${curr}`;
+    }
+    return acc;
+  }, '');
```

Comment:
No, because plugins can be nested.

`/x-pack/observability/public` is a valid plugin path, but so is `/x-pack/observability_solution/observability/public`.

Reply:
Right, but the code snippet I wrote would still work? Or am I missing something?

Reply:
`relativePathArray.slice(0, publicFolderIndex);` would return an array so it would need a `join('/')`. But after that, both implementations work the same.